controllers/apply-next/simple/fields.js

```diff
@@ -0,0 +1,682 @@
+        }
+    },
+    organisationAddressTownCity: {
+        name: `organisation-address-town-city`,
+        label: { en: 'Town or city', cy: '' },
+        type: 'text',
+        attributes: { size: 25 },
+        isRequired: true,
+        schema: Joi.string().required(),
+        messages: {
+            base: { en: 'Enter a town or city' }
+        }
+    },
+    organisationAddressCounty: {
+        name: `organisation-address-county`,
+        label: { en: 'County', cy: '' },
+        type: 'text',
+        attributes: { size: 25 },
+        isRequired: true,
+        schema: Joi.string().required(),
+        messages: {
+            base: { en: 'Enter a county' }
+        }
+    },
+    organisationAddressPostcode: postcodeField({
+        name: `organisation-address-postcode`,
+        label: { en: 'Postcode', cy: '' }
+    }),
+    organisationType: {
+        name: 'organisation-type',
+        label: { en: 'What type of organisation are you?', cy: '(WELSH) What type of organisation are you?' },
+        type: 'radio',
+        options: values(organisationTypes),
+        isRequired: true,
+        get schema() {
+            return Joi.string()
+                .valid(this.options.map(option => option.value))
+                .required();
+        },
+        messages: {
+            base: { en: 'Choose an organisation type' }
+        }
+    },
+    charityNumber: {
+        name: 'charity-number',
+        label: { en: 'Charity registration number', cy: '' },
+        explanation: {
+            en:
+                'If you are registered with OSCR, you only need to provide the last five digits of your registration number.',
+            cy: ''
+        },
+        type: 'text',
+        attributes: {
+            size: 20
+        },
+        isRequired: true,
+        schema: Joi.any().when('organsation-type', {
+            is: Joi.valid(organisationTypes.unincorporatedRegisteredCharity.value),
+            then: Joi.number().required()
+        }),
+        messages: {
+            base: { en: 'Enter a charity number' }
+        }
+    },
+    companyNumber: {
+        name: 'company-number',
+        label: { en: 'Companies house number', cy: '' },
+        type: 'text',
+        isRequired: true,
+        schema: Joi.any().when('organsation-type', {
+            is: Joi.valid([
+                organisationTypes.charitableIncorporatedOrganisation.value,
+                organisationTypes.notForProfitCompany.value,
+                organisationTypes.communityInterestCompany.value
+            ]),
+            then: Joi.string().required()
+        }),
+        messages: {
+            base: { en: 'Enter a companies house number' }
+        }
+    },
+    accountingYearDate: {
+        name: 'accounting-year-date',
+        label: { en: 'What is your accounting year end date?', cy: '' },
+        type: 'date',
+        isRequired: true,
+        schema: Joi.date().min('now'),
```

Comment:
Is this right? I think this logic is saying the accounting end date has to be today (or later), surely this should be the `max`? Also we should specify that value as an attribute for the client-side validation too, right?

Reply:
I think this is right, the accounting end date needs to be in the future, so needs to be _at least_ now?

Reply:
It does? I guess I'd assumed that you're submitting a summary of your accounts up to a certain (past) date? But fine, let's leave it as-is as I'm sure the phrasing/validation will evolve over time.

Reply:
Oh, I'd interpreted it completely differently, but you're right, we should leave it and confirm what the actual rule is.